### PR TITLE
backend setup and signup & login apis

### PR DESCRIPTION
**Pull request by mehtakevan**
Backend setup for the project is done. Also the signup/registration api and login api is created. The userModel is also been created using Mongodb.

### CODE REVIEW

**Comment by mehtakevan** [comment]
@abhay-raj19 Let me know if any other changes is required.

**Comment by abhay-raj19** [comment]
The backend is been added #107 you can add more apis for handling the login and signup functionality @mehtakevan

**Comment by mehtakevan** [comment]
@abhay-raj19 Okay sure I will add that apis.

File: server/models/User.js [hunk 1]
```diff
@@ -1,29 +1,51 @@
-/* This is a sample model file for the user schema. 
-This file will be used to create a user schema and export it to the server file. */
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
 
-import mongoose from 'mongoose'
+const userModel = mongoose.Schema({
+  id: { type: Number },
+  userName: { type: String },
+  password: { type: "String", required: true },
```
Comment (abhay-raj19) [inline server/models/User.js:7]: Separate all the schema models in the different folder models and export all those models from there and then use them inside the model

**Comment by mehtakevan** [comment]
@abhay-raj19 I have implemented the changes. Check it once.

**Comment by abhay-raj19** [comment]
LGTM

**Comment by abhay-raj19** [comment]
Thanks for contributing to Fitbody, Don't forget to star the repository ⭐.

**Comment by mehtakevan** [comment]
@abhay-raj19  can you please make me as assigned participate and label this under gssoc'24

**Pull request merged**